src/Http/Http.Abstractions/src/HttpContext.cs

```diff
@@ -71,4 +74,26 @@ public abstract class HttpContext
     /// Aborts the connection underlying this request.
     /// </summary>
     public abstract void Abort();
+
+    /// <inheritdoc />
+    public override string ToString()
+    {
+        return $"{Request.Method} {Request.Path.Value} {Request.ContentType}"
+            + $" Status: {Response.StatusCode} {Response.ContentType}";
+    }
+
+    private class HttpContextDebugView(HttpContext context)
```

Comment:
This is missing the `Feature` collection. I think we should always expose all public properties here.

In fact, it would be great if there were unit tests to verify the public properties on the debug views match the public properties on their underlying types. If we add a new property to HttpContext/HttpRequest/etc then there is a prompt to also add it to the debug view. Otherwise, I could imagine it being missed.

Reply:
I left it off on purpose because it doesn't have any useful public properties and the internal types are extremely complicated. I'll revisit trying to make some of the IFeatureCollection implementations pretty.

I also left off request and response bodies and readers/writers because they similarly have no useful public state, they exist only to call methods on. Response.Cookies is similar, only methods.

As for tests, I can check, but it's extremely rare to modify these types.

Reply:
There are three main implementations of IFeatureCollection.
- FeatureCollection (HttpSys, TestHost, and DefaultHttpContext use this)
- Kestrel
- IIS

FeatureCollection is easy enough, it's a real collection. Kestrel and IIS are problematic because the same type implements IFeatureCollection and most of the feature interfaces, so you end up with something silly like

| Features | Count = 12 |
| -- | -- |
| | ...IHttpRequestFeature, Count = 12 |
| | ...IHttpResponseFeature, Count = 12 |
| | ... |

The most reasonable view I can give for these cases is a flat list of feature names.
| Features | Count = 12 |
| -- | -- |
| | ...IHttpRequestFeature |
| | ...IHttpResponseFeature |
| | ... |